src/error_handling.c

```diff
@@ -236,13 +246,19 @@ rcutils_get_error_state(void)
 rcutils_error_string_t
 rcutils_get_error_string(void)
 {
-  if (!gtls_rcutils_error_is_set) {
+  if (!gtls_rcutils_error_is_set && !gtls_rcutils_error_code_is_set) {
     return (rcutils_error_string_t) {"error not set"};  // NOLINT(readability/braces)
   }
-  if (!gtls_rcutils_error_string_is_formatted) {
+  if (!gtls_rcutils_error_string_is_formatted && gtls_rcutils_error_is_set) {
     __rcutils_format_error_string(&gtls_rcutils_error_string, &gtls_rcutils_error_state);
     gtls_rcutils_error_string_is_formatted = true;
   }
+
+   if(!gtls_rcutils_error_code_is_formatted && gtls_rcutils_error_code_is_set){
+    __rcutils_format_add_error_code(&gtls_rcutils_error_string , &gtls_rcutils_error_code);
```

Comment:
if the condition with `gtls_rcutils_error_is_set` is false and `gtls_rcutils_error_code_is_set` is true, this will cause assertion error because `gtls_rcutils_error_string` could be NULL. (e.g. after calling `rcutils_reset_error`)

Reply:
Please see the updated code.